### PR TITLE
Fixes for Morello board purecap kernel bring-up

### PR DESCRIPTION
**Pull request by jrtc27**
Gathering any patches needed here; so far just the one, but more may come (i.e. don't merge this yet, let's batch them all up here as we find them). Deliberately branched off master to avoid issues with churn in dev.

### CODE REVIEW

**Comment by jrtc27** [comment]
Rebased, cleaned up the history and replaced upstreamed commits with cherry picks of my upstream commits. There will surely be other bugs found, but this is the set that was needed previously with the old version of master.

**Comment by bsdjhb** [comment]
Thanks, I will merge this after the next upstream merge.

**Review by brooksdavis — APPROVED** [review]
casual review seems fine

**Pull request merged**
